src/networking.c

```diff
@@ -1979,6 +1979,14 @@ void processInputBuffer(client *c) {
                 return;
             }
         }
+
+        /* If the client reached the soft limit, break so that we can write
+         * some data into the socket and process the rest of the pipeline
+         * on the next event. */
+        if (c->obuf_soft_limit_reached_time) {
+            queueClientForReprocessing(c);
+            break;
+        }
```

Comment:
Not for this PR, but some thoughts of mitigating eviction keys for mixing clients buffer to used memory
Can we also check all total clients buffer size (`stat_clients_type_memory[CLIENT_TYPE_NORMAL]`) at here, break early if it exceeds some one configuration we add.

Reply:
@ShooterIT thanks for looking into this. i'm working on such a mechanism for #7676. a new config limit, that will disconnect clients if the total usage (`stat_clients_type_memory`) is over a the limit, and prevent massive key eviction. it might also start by suspending some clients (like this PR does) in hope that it won't have to resort to client disconnection. (one of the challenges it to decide which client to suspend / disconnect).

it's still in early stages, but while working on it, and adding tests to reproduce the various scenarios that can trigger it, i came up with the mechanism in this PR and concluded it can be a PR of it's own.
i.e. it doesn't depend on a global limit and the combined memory usage, just considers each client on its own. and i'm also attempting to avoid adding any new per-client limit config (using the existing soft limit).

So while the mechanism of the back-pressure (delaying commands) can still be improved, my main question in this draft is if the idea of relying on the soft-limit, a good one?

Reply:
I see, my thoughts also are for #7676, your this PR inspired me. I think it is a good idea just I said above, currently default configuration of `client-output-buffer` is more safe after your PR, we just need tell users this clearly